test/ng/compileSpec.js

```diff
@@ -7815,4 +7817,125 @@ describe('$compile', function() {
         expect(element.hasClass('fire')).toBe(true);
       }));
   });
+
+  describe('element replacement', function() {
+    it('should broadcast $destroy only on removed elements, not replaced', function() {
```

Comment:
Interesting; just removing the `skipDestroyOnNextJQueryCleanData` logic from my commit definitely did fail unit tests.

This might be related, I totally forgot about this WIP PR: #8695